src/actions/promote.ts

```diff
@@ -41,4 +50,114 @@ export async function promote(ignoreSslErrors?: boolean) {
     } catch (ex) {
         core.warning('Exception occurred while deleting canary and baseline workloads. Exception: ' + ex);
     }
+}
+
+export async function promoteBlueGreen(ignoreSslErrors?: boolean) {
+    const kubectl = new Kubectl(await utils.getKubectl(), TaskInputParameters.namespace, ignoreSslErrors);
+
+    if (!blueGreenDeploymentHelper.isBlueGreenDeploymentStrategy()) {
+        core.debug('Strategy is not blue-green deployment. Invalid request.');
+        throw ('InvalidPromotetActionDeploymentStrategy');
+    }
+
+    let inputManifestFiles: string[] = getManifestFiles(TaskInputParameters.manifests);
+
+    inputManifestFiles = updateContainerImagesInManifestFiles(inputManifestFiles, TaskInputParameters.containers);
+
+    // imagePullSecrets addition
+    inputManifestFiles = updateImagePullSecretsInManifestFiles(inputManifestFiles, TaskInputParameters.imagePullSecrets);

```

Comment:
This is duplicate code across the entire action; consider moving it to a util file